### PR TITLE
Move metadata to pyproject.toml

### PR DESCRIPTION
**Pull request by cdce8p**
## Proposed change
Similar to https://github.com/home-assistant/core/pull/72422
Move metadata keys from `setup.cfg` to `pyproject.toml`.

Depends on https://github.com/home-assistant/actions/pull/71

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: script/version_bump.js [hunk 1]
```diff
@@ -50,14 +50,14 @@ async function main(args) {
     return;
   }
 
-  const setup = fs.readFileSync("setup.cfg", "utf8");
+  const setup = fs.readFileSync("pyproject.toml", "utf8");
   const version = setup.match(/\d{8}\.\d+/)[0];
```
Comment (ludeeus) [inline script/version_bump.js:54]:
```suggestion
  const version = setup.match(/version\W+=\W"(\d{8}\.\d)"/)[1];
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -28,6 +28,8 @@ jobs:
 
       - name: Verify version
         uses: home-assistant/actions/helpers/verify-version@master
+        with:
+          metadata-file: pyproject.toml
```
Comment (ludeeus) [inline .github/workflows/release.yaml:32]: Not too sure about that, why does not the action just handle it regardless of which of the meta file types that are used?

Comment (cdce8p) [reply .github/workflows/release.yaml]: Good point. I was overcomplicating things. Updated https://github.com/home-assistant/actions/pull/71 as well.

**Review by ludeeus — APPROVED** [review]
✅  Needs to wait for the action to be updated before merging.

**Comment by cdce8p** [comment]
> ✅ Needs to wait for the action to be updated before merging.

I think it can be merged now.

**Pull request merged**